### DISCUSSION TITLE
No space left on device when deploying

### DISCUSSION THREAD

**Discussion by fredrikst** [aws/sagemaker-python-sdk#4019] [Help]
I'm using HuggingfaceModel to create the model and deploy an endpoint. The instance type is ml.g5.x2large, which clearly states 450GB storage space in the specs. The model size is around 30GB.

What am I missing here, do I need to specify a certain cache_dir?

**Comment by pottieral** [comment]
Hi, I am facing the same problem, were you able to solve it?

**Comment by Wingjam** [comment]
Im also facing the same issue with the following code:

```python
from sagemaker.huggingface import HuggingFaceModel

huggingface_model = HuggingFaceModel(
    model_data=s3_model_uri,
    role=role,
    transformers_version="4.26.0",
    pytorch_version="1.13.1",
    py_version="py39",
    entry_point="inference.py",
)

predictor = huggingface_model.deploy(
    initial_instance_count=1,
    instance_type="ml.g4dn.8xlarge",
)

print(f"Endpoint Name: {predictor.endpoint_name}")
```

Got this error: `OSError: [Errno 28] No space left on device`
